src/groupByKey.ts

```diff
@@ -2,6 +2,18 @@ type GroupsMap<T> = {
   [key: string]: T[];
 };
 
-export function groupByKey(items, key) {
-  // write code here;
+export function groupByKey<T>(items: T[], key: keyof T): GroupsMap<T> {
+  const result: GroupsMap<T> = {};
+
+  items.forEach((item) => {
+    const keyValue = item[key] as string;
+
+    if (!result[keyValue]) {
+      result[keyValue] = [item];
+    } else {
+      result[keyValue].push(item);
+    }
```

Comment:
make sense to use ternary operator here as it will make a code shorter

Reply:
I think this code is quite short
There will be even more code to fix